quiche/src/recovery/gcongestion/bbr2/network_model.rs

```diff
@@ -630,21 +630,30 @@ impl BBRv2NetworkModel {
 
     pub(super) fn check_persistent_queue(
         &mut self, target_gain: f32, params: &Params,
-    ) {
+    ) -> PersistentQueueOutcome {
+        let mut outcome = PersistentQueueOutcome {
+            full_bandwidth_reached: false,
+            rounds_with_queueing: 0,
+        };
+
         let target = self
             .bdp(self.max_bandwidth(), target_gain)
             .max(self.bdp0() + self.queueing_threshold_extra_bytes());
 
         if self.min_bytes_in_flight_in_round < target {
             self.rounds_with_queueing = 0;
-            return;
-        }
+        } else {
+            self.rounds_with_queueing += 1;
 
-        self.rounds_with_queueing += 1;
-        #[allow(clippy::absurd_extreme_comparisons)]
-        if self.rounds_with_queueing >= params.max_startup_queue_rounds {
-            self.full_bandwidth_reached = true;
+            if self.rounds_with_queueing >= params.max_startup_queue_rounds {
+                self.full_bandwidth_reached = true;
+                outcome.full_bandwidth_reached = true;
+            }
         }
+
+        outcome.rounds_with_queueing = self.rounds_with_queueing;
+
+        outcome
```

Comment:
Refactors in this code make me uneasy given the testing state of the code in the gcongestion directory.

At the very least no-op refactors should be in separate commits which can be inspected independent of code changes that introduce new behaviors.

I don't think that this refactor is related to the introduction of startup related stats.  I strongly suggest we undo all changes to this function.